src/ripple/app/main/Application.cpp

```diff
@@ -1034,6 +1034,35 @@ class ApplicationImp
                     << "Remaining free disk space is less than 512MB";
                 signalStop ();
             }
+
+            static std::atomic<std::uint32_t> pageSize {};
+            static std::atomic<std::uint32_t> maxPages;
+            auto db = mTxnDB->checkoutDb();
+            if (!pageSize)
+            {
+                std::uint32_t ps, mp;
+                *db << "PRAGMA page_size;", soci::into(ps);
+                *db << "PRAGMA max_page_count;" , soci::into(mp);
+                pageSize = ps;
+                maxPages = mp;
+            }
+            std::uint32_t pageCount;
+            *db << "PRAGMA page_count;", soci::into(pageCount);
+            if ((maxPages - pageCount) * pageSize < bytes512M)
+            {
+                std::stringstream ss;
+                ss
+                    << "Free SQLite space for transaction db is less than "
+                       "512MB. To fix this, rippled must be executed with the "
+                       "vacuum <tmpdir> parameter before restarting. "
+                       "Note that this activity can take multiple days, "
+                       "depending on database size."
+                       "page_size count max: "
+                    << pageSize << " " << pageCount << " " << maxPages;
```

Comment:
A `stringstream` isn't necessary, you can stream into the journal.
eg.
```
JLOG(m_journal.fatal()) <<
    "Free SQLite space for transaction db is less than "
    "512MB. To fix this, rippled must be executed with the "
    "vacuum <tmpdir> parameter before restarting. "
    "Note that this activity can take multiple days, "
    "depending on database size."
    "page_size count max: " <<
    pageSize << " " <<
    pageCount << " " <<
    maxPages;
```

Reply:
Fixed. As I was writing that code, I was thinking to myself "wouldn't it be nice if the journal object did this."